controllers/channel.js

```diff
@@ -1,44 +1,56 @@
 'use strict';
 
 var config = require(__dirname + '/../config/config'),
-    mysql = require('anytv-node-mysql');
+    mysql = require('anytv-node-mysql'),
+    util = require(__dirname + '/../helpers/util');
 
 exports.get_channels = function (req, res, next) {
     var limit = 50,
         offset = 0,
         total = 0,
         page = 1,
+        data = util.get_data([], ['page', 'per_page', 'q'], req.query);
 
-    start = function () {
-        var query = '';
+    function start () {
+        var query;
 
-        if (+req.query.page && +req.query.page > 0) {
-            page = +req.query.page;
+        query = 'SELECT count(channel_id) as aggregate FROM mcn_channels'
+                    + ' WHERE (linked_date != "" OR linked_date is not null)'
+                    + ' AND network != "" AND is_terminated != 1';
+
+        if (+data.page && +data.page > 0) {
```

Comment:
do not assign to a new `page` variable, use `data.page` instead`. 

`data.page = (+data.page > 0) ? +data.page: <default_page>;`